android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -146,7 +148,21 @@ constructor(
     questionnaireConfig: QuestionnaireConfig,
   ): Questionnaire? {
     if (questionnaireConfig.id.isEmpty() || questionnaireConfig.id.isBlank()) return null
-    return defaultRepository.loadResource<Questionnaire>(questionnaireConfig.id)
+    var result =
+      ContentCache.getResource(ResourceType.Questionnaire.name + "/" + questionnaireConfig.id)
+        ?.copy()
```

Comment:
I wonder if we should move the `.copy(` operation to the Content cache itself, that way the caller only has to pass in the resource ID and not worry about creating a deep copy which they might forget to do and might lead to a few issues.